src/go/cmd/customer-trace-tester/main.go

```diff
@@ -0,0 +1,269 @@
+
+	return bsp
+}
+
+func configureJaegerThriftHttpExporter(collectorHostName string) sdktrace.SpanProcessor {
+	url := fmt.Sprintf("http://%s:14268/api/traces", collectorHostName)
+	log.Printf("Jaeger Thrift HTTP Exporter url: %s\n", url)
+
+	traceExporter, err := jaeger.New(
+		jaeger.WithCollectorEndpoint(
+			jaeger.WithEndpoint(url),
+		),
+	)
+	handleErr("Failed to create Zipkin trace exporter", err)
+
+	bsp := sdktrace.NewBatchSpanProcessor(traceExporter)
+
+	return bsp
+}
+
+func initProvider(ctx context.Context, spanProcessor sdktrace.SpanProcessor) func() {
+	res, err := resource.New(ctx,
+		resource.WithAttributes(
+			// the service name used to display traces in backends
+			semconv.ServiceNameKey.String("customer-trace-test-service"),
+		),
+	)
+	handleErr("failed to create resource", err)
+
+	// Register the trace exporter with a TracerProvider, using a batch
+	// span processor to aggregate spans before export.
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+		sdktrace.WithResource(res),
+		sdktrace.WithSpanProcessor(spanProcessor),
+	)
+	otel.SetTracerProvider(tracerProvider)
+
+	// set global propagator to tracecontext (the default is no-op).
+	otel.SetTextMapPropagator(propagation.TraceContext{})
+
+	return func() {
+		// Shutdown will flush any remaining spans and shut down the exporter.
+		handleErr("failed to shutdown TracerProvider", tracerProvider.Shutdown(ctx))
+	}
+}
+func waitForAWhile() {
+	max := 1000
+	min := 10
+	time.Sleep(time.Duration(rand.Intn(max-min+1)+min) * time.Millisecond)
+}
+
+func buildSpan(tracer trace.Tracer, parentCtx context.Context, countNumber int) (context.Context, trace.Span) {
+	ctx, childSpan := tracer.Start(
+		parentCtx,
+		fmt.Sprintf("child-%d", countNumber+1),
+		trace.WithAttributes(
+			attribute.Int("countNumber", countNumber+1),
+		),
+	)
+
+	childSpan.SetName(fmt.Sprintf("ancestor-%d", countNumber+1))
+	return ctx, childSpan
+}
+
+func buildTrace(tracer trace.Tracer, ctx context.Context, spansPerTrace int, name string) {
+	ctx, parentSpan := tracer.Start(ctx, "parent")
+	parentSpan.SetName(fmt.Sprintf("root-span-%s", name))
+	waitForAWhile()
+	parentSpan.End()
+
+	currentCtx := ctx
+	for i := 0; i < spansPerTrace-1; i++ {
+		newCtx, childSpan := buildSpan(tracer, currentCtx, i)
+		currentCtx = newCtx
+		waitForAWhile()
+		childSpan.End()
+	}
+}
+
+func runCustomerTraceTest(testCfg traceTestConfig, name string) {
+	tracer := otel.Tracer(name)
+
+	tracesCount := testCfg.totalTraces
+	spansPerTrace := testCfg.spansPerTrace
+	for i := 0; i < tracesCount; i++ {
+		buildTrace(tracer, context.Background(), spansPerTrace, name)
+	}
+}
+
+func handleErr(message string, err error) {
+	if err != nil {
+		log.Fatalf("%s: %v\n", message, err)
+	}
+}
+
+var help bool
+
+func init() {
+	flag.BoolVar(&help, "help", false, "show help")
+}
+
+func main() {
+	flag.Parse()
+	if help {
+		fmt.Println("Customer trace-testing")
```

Comment:
Can we add some more info here? I'd expect actual "help" when I run the binary with `-help` flag :)

Reply:
Done :)